src/app/lib/views/browser/list.js

```diff
@@ -237,25 +233,36 @@
         },
 
         itemsPerRow: function () {
-            var total = 0;
-            var items = $(document.querySelector('.items')).children('.item');
+            var currentWidth = this.$el.width();
+            var minPerRow = Math.floor(1 / (Settings.postersWidth / currentWidth) - 1);
+
+            // recalc only if needed
+            if (
+                currentWidth !== this.ui.row.listWidth // win was resized?
+                || Settings.postersWidth !== this.ui.row.posterWidth // poster were resized?
+                || this.ui.row.itemsPerRow < minPerRow // something impossible happened?
```

Comment:
I disagree. It would duplicate the return and would change the conditions to && making it actually harder to process, because all conditions will need to be checked.

Test conditions yourself: 
```
var a = function (s) {console.log('fn a'); return s;};
var b = function (s) {console.log('fn b'); return s;};
var c = function (s) {console.log('fn c'); return s;};
if (!a(false) || !b(false) || !c(false)) console.log('done');
```